src/plugin.js

```diff
@@ -41,70 +35,113 @@ export default class Plugin {
 	}
 
 	/**
-	 * An array of plugins required by this plugin.
-	 *
-	 * To keep a plugin class definition tight it's recommended to define this property as a static getter:
-	 *
-	 *		import Image from './image.js';
-	 *
-	 *		export default class ImageCaption extends Plugin {
-     *			static get requires() {
-     *				return [ Image ];
-     *			}
-	 *		}
-	 *
-	 * @static
-	 * @readonly
-	 * @member {Array.<Function>|undefined} module:core/plugin~Plugin.requires
-	 */
-
-	/**
-	 * Optional name of the plugin. If set, the plugin will be available in
-	 * {@link module:core/plugincollection~PluginCollection#get} by its
-	 * name and its constructor. If not, then only by its constructor.
-	 *
-	 * The name should reflect the package name + the plugin module name. E.g. `ckeditor5-image/src/image.js` plugin
-	 * should be named `image/image` (the `ckeditor5-` prefix is stripped during compilation). If plugin is kept
-	 * deeper in the directory structure, it's recommended to only use the module file name, not the whole path.
-	 * So, e.g. a plugin defined in `ckeditor5-ui/src/notification/notification.js` file may be named `ui/notification`.
-	 *
-	 * To keep a plugin class definition tight it's recommended to define this property as a static getter:
-	 *
-	 *		export default class ImageCaption {
-     *			static get pluginName() {
-     *				return 'image/imagecaption';
-     *			}
-	 *		}
-	 *
-	 * @static
-	 * @readonly
-	 * @member {String|undefined} module:core/plugin~Plugin.pluginName
-	 */
-
-	/**
-	 * The second stage (after plugin {@link #constructor}) of plugin initialization.
-	 * Unlike the plugin constructor this method can perform asynchronous.
-	 *
-	 * A plugin's `init()` method is called after its {@link module:core/plugin~Plugin.requires dependencies} are initialized,
-	 * so in the same order as constructors of these plugins.
-	 *
 	 * @returns {null|Promise}
 	 */
 	init() {}
 
 	/**
-	 * The third (and last) stage of plugin initialization. See also {@link #constructor} and {@link #init}.
-	 *
 	 * @returns {null|Promise}
 	 */
 	afterInit() {}
 
 	/**
-	 * Destroys the plugin.
-	 *
 	 * @returns {null|Promise}
 	 */
 	destroy() {}
 }
 
 mix( Plugin, ObservableMixin );
+
+/**
+ * The base interface for CKEditor plugins.
+ *
+ * @interface Plugin
```

Comment:
I would call it `Plugin` and the base plugin class `BasePlugin` (which implements the `Plugin` interface).